src/ahbicht/expressions/expression_resolver.py

```diff
@@ -0,0 +1,46 @@
+"""
+This module makes it possible to parse expressions with all its subexpressions
+for example ahb_expressions with condition_expressions or condition_expressions with packages.
+"""
+
+from lark import Transformer, Tree
+
+from ahbicht.expressions.ahb_expression_parser import parse_ahb_expression_to_single_requirement_indicator_expressions
+from ahbicht.expressions.condition_expression_parser import parse_condition_expression_to_tree
+
+
+def expression_resolver(expression: str, resolve_packages: bool = False) -> Tree:
+    """
+    Parses expressions and resolves its subexpressions,
+    for example condition_expressions in ahb_expressions or packages in condition_expressions.
+    :param expression: a syntactically valid ahb_expression or condition_expression
+    :param resolve_packages: if true resolves also the packages in the condition_expressions
+    """
+    # TOOD: implement packages
+    if resolve_packages:
+        raise NotImplementedError("Resolving Packages is not implemented yet.")
+    try:
+        expression_tree = parse_ahb_expression_to_single_requirement_indicator_expressions(expression)
+        resolved_expression_tree = AhbExpressionResolverTransformer().transform(expression_tree)
+    except SyntaxError as ahb_syntax_error:
+        try:
+            resolved_expression_tree = parse_condition_expression_to_tree(expression)
+        except SyntaxError as condition_syntax_error:
+            # pylint: disable=raise-missing-from
+            raise SyntaxError(f"{ahb_syntax_error.msg} {condition_syntax_error.msg}")
```

Comment:
We can totally discuss other ways to raise the errors here, if you have a better idea.
Other ways I thought of would be to raise one from the other or have a totally new error message.

Reply:
Wie wäre es mit der error message:
> f"The expression '{expression}' is neither an AHB expression ({ahb_syntax_error.msg}) nor a condition expression ({condition_syntax_error.msg}).

Reply:
Die beiden Error-Messages sind halt sehr lang, deswegen ist es glaube ich schwer einen ganzen Satz daraus zu bilden:
>Please make sure that the ahb_expression starts with a requirement indicator (i.e Muss/M, Soll/S, Kann/K, X, O, U) and the condition expressions consist of only the following characters: [ ] ( ) U ∧ O ∨ X ⊻ and digits.

>Please make sure that:
             * all conditions have the form [INT]
             * no conditions are empty
             * all compositions are combined by operators 'U'/'O'/'X' or without an operator
             * all open brackets are closed again and vice versa